programs/test/CMakeLists.txt

```diff
@@ -16,7 +16,8 @@ target_link_libraries(benchmark PRIVATE ${tfpsacrypto_target} ${CMAKE_THREAD_LIB
 
 # which_aes
 tfpsacrypto_build_program_common(which_aes)
-target_link_libraries(which_aes PRIVATE ${tfpsacrypto_target})
+target_sources(which_aes PRIVATE $<TARGET_OBJECTS:tf_psa_crypto_test>)
```

Comment:
I think it would be better to not do that as this rather hides than fixes the issue. The core issue seems that we have two separate libraries, in the C compiler/linker sense, for the PSA core and the driver builtin that reference each other and it is linker dependent how this is handled. We should probably have only one library and in the CMake scripts to define the builtin library as an object library. I will create an issue for that.

Reply:
Here is the issue: https://github.com/Mbed-TLS/TF-PSA-Crypto/issues/300